### PR TITLE
Sw 249 condition handlers productnumbersearch

### PR DESCRIPTION
**Pull request by zep-com**

### CODE REVIEW

File: FinSearchUnified/Tests/Bundle/SearchBundleFindologic/ConditionHandler/CategoryConditionHandlerTest.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    /**
+     * @dataProvider categoryIdsDataProvider
+     *
+     * @param array $categoryIds
+     * @param array $expectedCategoryNames
+     *
+     * @throws Exception
+     */
+    public function testGenerateCondition($categoryIds, $expectedCategoryNames)
```
Comment (ihanli) [inline FinSearchUnified/Tests/Bundle/SearchBundleFindologic/ConditionHandler/CategoryConditionHandlerTest.php]: Add the missing typehints to the argument list.

File: FinSearchUnified/Tests/Bundle/SearchBundleFindologic/ConditionHandler/CategoryConditionHandlerTest.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+    public function categoryIdsDataProvider()
+    {
+        return [
+            'Single ID of a category without parents' => [[5], ['Genusswelten']],
+            'One Category without parents and one category having parents' => [
+                [5, 12],
+                ['Genusswelten', 'Genusswelten_Tees und Zubehör_Tees']
+            ],
+            'Category ID of "1"' => [[1], []],
```
Comment (ihanli) [inline FinSearchUnified/Tests/Bundle/SearchBundleFindologic/ConditionHandler/CategoryConditionHandlerTest.php]: I actually thought that Shopware would give you an empty string when you try to get the path for "Root". It seems that this is not 100% true.

So here's what are going to do:
* Remove the explicit handling of category with ID 1 from the condition handler.
* Use category "Deutsch" instead (should have the ID 3).

That should accurately represent what we are trying to do.

File: FinSearchUnified/Tests/Bundle/SearchBundleFindologic/ConditionHandler/PriceConditionHandlerTest.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+    /**
+     * @dataProvider priceDataProvider
+     *
+     * @param array $prices
+     * @param array $expectedPrices
+     *
+     * @throws Exception
+     */
+    public function testGenerateCondition($prices, $expectedPrices)
```
Comment (ihanli) [inline FinSearchUnified/Tests/Bundle/SearchBundleFindologic/ConditionHandler/PriceConditionHandlerTest.php]: Add the missing typehints to the argument list.

File: FinSearchUnified/Tests/Bundle/SearchBundleFindologic/ConditionHandler/ProductAttributeConditionHandlerTest.php [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+    /**
+     * @dataProvider attributesDataProvider
+     *
+     * @param array $attributes
+     * @param array $expectedValues
+     *
+     * @throws Exception
+     */
+    public function testGenerateCondition($attributes, $expectedValues)
```
Comment (ihanli) [inline FinSearchUnified/Tests/Bundle/SearchBundleFindologic/ConditionHandler/ProductAttributeConditionHandlerTest.php]: Add the missing typehints to the argument list.

File: FinSearchUnified/Bundle/SearchBundleFindologic/ConditionHandler/CategoryConditionHandler.php [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+     * @param QueryBuilder $query
+     * @param ShopContextInterface $context
+     *
+     * @throws Exception
+     */
+    public function generateCondition(ConditionInterface $condition, QueryBuilder $query, ShopContextInterface $context)
+    {
+        $categories = [];
+
+        /** @var CategoryCondition $condition */
```
[the remark this answers is not in the document]
Comment (zep-com) [reply FinSearchUnified/Bundle/SearchBundleFindologic/ConditionHandler/CategoryConditionHandler.php]: It is not needed as such as it is just a warning, but these methods are specific to the conditions and not available in the interface so it cannot directly get these without the type comment

File: FinSearchUnified/Bundle/SearchBundleFindologic/ConditionHandler/CategoryConditionHandler.php [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        foreach ($condition->getCategoryIds() as $categoryId) {
+            if ($categoryId === 1) {
+                continue;
+            }
+            $categoryName = StaticHelper::buildCategoryName($categoryId, false);
+            if (!empty($categoryName)) {
+                $categories[] = $categoryName;
+            }
+        }
+        $query->addCategories($categories);
```
[the remark this answers is not in the document]
Comment (ihanli) [reply FinSearchUnified/Bundle/SearchBundleFindologic/ConditionHandler/CategoryConditionHandler.php]: I don't see any change in this line (see previous comment).

[the remark this answers is not in the document]
Comment (zep-com) [reply FinSearchUnified/Bundle/SearchBundleFindologic/ConditionHandler/CategoryConditionHandler.php]: I actually had done this, but then I had kind of re-based the SW-250 branch and tried pushing it but it rejected the push saying it needs tpo be merged first. So I had to revert the rebase again and I might have missed it the 2nd time.

**Pull request merged**